src/iam.ts

```diff
@@ -1332,36 +1357,43 @@ export class IAM extends IAMBase {
   }
 
   async issueClaimRequest({
-    requesterDID,
+    requester,
+    token,
     id,
-    token
   }: {
-    requesterDID: string;
+    requester: string;
     token: string;
     id: string;
   }) {
     if (!this._did) {
       throw new Error(ERROR_MESSAGES.USER_NOT_LOGGED_IN);
     }
 
+    const { sub } = await this.decodeJWTToken({ token }) as { sub: string };
+    if (requester !== sub &&
+      !(await this._cacheClient.getOwnedAssets({ did: requester })).some((a) => a.document.id === sub)
+    ) {
+      throw new Error(ERROR_MESSAGES.CLAIM_REQUESTER_NOT_AUTHORIZED);
+    }
+
```

Comment:
@marcin-l-tsh I believe this is the check that is needed for https://energyweb.atlassian.net/browse/SWTCH-809
(does that seem right @JGiter ?)

Reply:
Right. I will remove it from here then to not add delay to request owned assets